EIPS/eip-3224.md

```diff
@@ -0,0 +1,447 @@
+  Described Data:     0xf14fcbc8e4a4f2bb818545497be34c7ab30e6e87e0001df4ba82e7c8b3f224fbaf255b91
+  i.e.                encodeWithSelector("commit(bytes32)", makeCommitment("ricmoose", "0xE31f43C1d823AfAA67A8C5fbB8348176d225A79e", "0x65462b0520ef7d3df61b9992ed3bea0c56ead753be7c8b3614e0ce01e4cac41b"))
+
+Signing:
+  Signed Transaction: 0xf88180808094ab3045aa85cbcabb06ed3f3fe968fa545772727080a4f14fcbc8e4a4f2bb818545497be34c7ab30e6e87e0001df4ba82e7c8b3f224fbaf255b912aa0a62b41d1ebda584fe84cf8a05f61b429fe4ec361e13c17f30a23281106b38a8da00bcdd896fe758d8f0cfac46445a48f76f5e9fe27790d67c51412cb98a12a0844
+```
+
+**Example: WETH mint()**
+
+```
+Input:
+  Address:            0xab3045AA85cBCaBb06eD3F3FE968fA5457727270
+  Describer Input:    0x1249c58b00000000000000000000000000000000000000000000000000000000
+  i.e.                encode(
+                          [ "bytes4" ],
+                          [ SEL("mint()") ]
+                      )
+  Value:              1.23 ether
+
+Output:
+  Description:        text/plain;Mint 1.23 WETH (spending 1.23 ether)?
+  Described Data:     0x1249c58b
+  i.e.                encodeWithSelector("mint()")
+
+Signing:
+  Signed Transaction: 0xf86980808094ab3045aa85cbcabb06ed3f3fe968fa5457727270881111d67bb1bb0000841249c58b29a012df802e1394a97caab23c15c3a8c931668df4b2d6d604ca23f3f6b836d0aafca0071a2aebef6a9848616b4d618912f2003fb4babde3dba451b5246f866281a654
+```
+
+## Reference Implementation
+
+@TODO (consider adding it as one or more files in `../assets/eip-####/`)
+
+I will add examples in Solidity and JavaScript.
+
+
+## Security Considerations
+
+### Escaping Text
+
+Wallets must be careful when displaying text provided by
+contracts and proper efforts must be taken to sanitize
+it, for example, be sure to consider:
+
+- HTML could be enbedded to attempt to trick web-based wallets into [executing code](https://en.wikipedia.org/wiki/Code_injection) using the script tag (possibly uploading any private keys to a server)
+- In general, extreme care must be used when rendering HTML; consider the ENS names `<span style="display:none">not-</span>ricmoo.eth` or `&thinsp;ricmoo.eth`, which if rendered without care would appear as `ricmoo.eth`, which it is not
+- Other marks which require escaping could be included, such as quotes (`"`), formatting (`\n` (new line), `\f` (form feed), `\t` (tab), any of many [non-standard whitespaces](https://en.wikipedia.org/wiki/Whitespace_character#Spaces_in_Unicode)), back-slassh (`\`)
+- UTF-8 has had bugs in the past which could allow arbitrary code execution and [crashing renderers](https://osxdaily.com/2015/05/27/bug-crashes-messages-app-ios-workaround/); consider using the UTF-8 replacement character (or *something*) for code-points outside common planes or common sub-sets within planes
+- [Homoglyphs attacks](https://en.wikipedia.org/wiki/IDN_homograph_attack)
+- [Right-to-left](https://en.wikipedia.org/wiki/Right-to-left_mark) mark may affect rendering
```

Comment:
External links should be avoided whenever possible, as they tend to 404 over time and we don't have the manpower to maintain dead links.  I think most of these are easily searchable, so no need to link to them (naming them is fine).